src/compiler/builder.ts

```diff
@@ -1214,6 +1221,17 @@ function getBuildInfo(state: BuilderProgramState): BuildInfo {
         return root.length = root.length - 1;
     }
 
+    function toResolvedRoot(): ProgramBuildInfoResolvedRoot[] | undefined {
+        let result: ProgramBuildInfoResolvedRoot[] | undefined;
+        rootFileNames.forEach(path => {
+            const file = state.program!.getSourceFileByPath(path);
+            if (file && path !== file.resolvedPath) {
+                (result ??= []).push([toFileId(file.resolvedPath), toFileId(path)]);

```

Comment:
I don't think this pattern is new in the file, but it does seem like we could be using `append` across the board.